pkg/kube/discover.go

```diff
@@ -40,6 +40,21 @@ func IsOSRouteGroupAvailable(ctx context.Context, cli discovery.DiscoveryInterfa
 	return false, nil
 }
 
+func IsResAvailableInGroupVersion(ctx context.Context, cli discovery.DiscoveryInterface, groupName, version, resource string) (bool, error) {
+	resList, err := cli.ServerPreferredResources()
+	if err != nil {
+		return false, err
+	}
+	for _, res := range resList {
+		for _, r := range res.APIResources {
+			if r.Name == resource && fmt.Sprintf("%s/%s", groupName, version) == res.GroupVersion {
```

Comment:
I think, we can have a minor improvement here: `fmt.Sprintf("%s/%s", groupName, version)` is called multiple times, but always evaluates to the same value.

Reply:
Hi @denisvmedia,
Sorry, I didn't get that. `fmt.Sprintf("%s/%s", groupName, version)` will result into different values in every iteration of the loop.

Reply:
aah, hold on.

Reply:
Hm... are you sure? `groupName` and `version` are passed as string arguments in the function. Therefore they seem to remain unchanged for every iteration of the loop.